src/mcts/search.cc

```diff
@@ -515,13 +536,20 @@ void SearchWorker::GatherMinibatch() {
 
     // If node is already known as terminal (win/loss/draw according to rules
     // of the game), it means that we already visited this node before.
-    if (node->IsTerminal()) continue;
+    // Furthermore, we must pass this termination value to NodeToProcess::v,
+    // both here and after ExtendNode.
+    if (node->IsTerminal()) {
+      nodes_to_process_.back().v = node->GetTerminalNodeValue();
```

Comment:
It feels to be wrong place to fill `v`.
Kind of wrong that v comes from different places for terminal and non terminal nodes. It's easy to remove one and forget another.

`FetchNNResults` really feels the place "where V is populated", but then it probably should be renamed to `PopulateEvalsIntoNodes` or something like that.

Also in this function, code duplication..

Reply:
Your thoughts mirror mine. Perhaps instead of `FetchNNResults` it could be `FetchMinibatchResults`, including both actually-network'd nodes and nodes that weren't.